vars/promotionHelper.groovy

```diff
@@ -29,22 +33,29 @@ void abortPreviousBuilds(RunWrapper promotedBuild) {
  *
  * @param promotedBuild the downstream build instance to be queried
  * @param upstreamJobName the upstream build name
+ * @param instance Hudson instance as returned by `Jenkins.instance`. Mostly useful to override when testing.
```

Comment:
If this variable is only added for the sake of the tests then I think it would be better to mock "Jenkins.getInstanceOrNull()" within the tests and get rid of this variable. That should be possible using the test framework but erg test frameworks 🤢 .

Reply:
It wasn't really possible until I updated the mocking framework to be PowerMock, and now seems like it's simple.

There are a couple of arguments that it:

1. the instance is kind of always there so it might already be assigned, so this could constitute re-retrieving something you already have (in some cases)
1. the function isn't pulling in global state if it's passed as a parameter

That said, I think it's probably more natural in the context of a Jenkins shared library to just access `Jenkins.instance` when you want it, so I've mocked it in the tests to keep the actual code more natural and uncluttered with test stuff.

Reply:
https://github.com/aodn/jenkins-shared-library/pull/15/files#diff-c9c050b37fc9946998aaa1c6db5d8629R38

As described at: https://wiki.jenkins.io/display/JENKINS/Mocking+in+Unit+Tests